lib/openc_bot/helpers/register_methods.rb

```diff
@@ -167,17 +167,19 @@ def schema_name
       end
 
       def stale_entry_uids(stale_count = nil)
-        stale_count ||= default_stale_count
-        sql_query = "ocdata.#{primary_key_name} from ocdata WHERE retrieved_at IS NULL OR strftime('%s', retrieved_at) < strftime('%s',  '#{Date.today - days_till_stale}') order by datetime( retrieved_at ) LIMIT #{stale_count.to_i}"
-        select(sql_query).each do |res|
-          yield res[primary_key_name.to_s]
+        handle_retrieved_at_not_exists do
+          stale_count ||= default_stale_count
+          sql_query = "ocdata.#{primary_key_name} from ocdata WHERE retrieved_at IS NULL OR strftime('%s', retrieved_at) < strftime('%s',  '#{Date.today - days_till_stale}') order by datetime( retrieved_at ) LIMIT #{stale_count.to_i}"
+          select(sql_query).each do |res|
+            yield res[primary_key_name.to_s]
+          end
         end
-      rescue SQLite3::SQLException => e
-        if e.message[/no such column: retrieved_at/]
-          sqlite_magic_connection.add_columns("ocdata", ["retrieved_at"])
-          retry
-        else
-          raise e
+      end
+
+      def assess_stale
+        handle_retrieved_at_not_exists do
+          sql_query = "count(*) from ocdata WHERE retrieved_at < '#{Date.today - days_till_stale}'"
```

Comment:
I'd expect this to use the exact same where clause as the query above (so you could define a constant STALE_SQL_CLAUSE) 
Looks like the query above has some extra robustness around retrieved_at being formatted wrong, and I suppose we'd want to include those where retrieved_at is null in the count, since they're included in the processing. Or did you decide not to include those?

`STALE_SQL_CLAUSE="retrieved_at IS NULL OR strftime('%s', retrieved_at) < strftime('%s',  '#{Date.today - days_till_stale}')"`

Reply:
Originally I was thinking the formatting of date is pointless considering we are standardising such input and it would only effect the ordering of return, but it would not hurt and I'm sure there are some race conditions I haven't thought about that would be a gotcha.

RE the NULL, it's now impossible for companies without a retrieved_at to make their way in to the sqlite and they would throw errors on ingest. I don't think we would want to count any company record with a NULL retrieved_at to any metric as it is indicative of something very wrong, at least from a data P.O.V